### PR TITLE
Metrics panel from Network Navigator

### PR DESCRIPTION
**Pull request by jrladd**
This fork creates a "Metrics" panel, right below settings, that mimics the functions of [Network Navigator](https://dh-web.hss.cmu.edu/network_navigator/).

Changes include:

- styling and template for the panel
- functions for calculating network metrics
- a test for the lookup file to ensure that the network is unimodal
- a warning message if the network *isn't* unimodal
- addition of jsnetworkx to power the network functions

### CODE REVIEW

File: palladio-graph-view.css [hunk 1]
```diff
@@ -80,3 +80,59 @@
 [data-palladio-graph-view] .link.hidden {
     display: none;
 }
+
+.metrics {
```
Comment (esjewett) [inline palladio-graph-view.css]: @jrladd Can you prefix all of these CSS class selectors with `[data-palladio-graph-view]`? If we don't do this, they could conflict with other classes in other components.

**Comment by esjewett** [comment]
I'd like to get @simonwiles input on whether it makes sense to try to convert this to a webpack build and use the jsnetworkx NPM library. Since I'm guessing we can't do that in a short period of time, I'm good to merge this as-is with the following changes:

* Update the CSS file as described in the comment and rebuild
* Update the LICENSE file to include (at the bottom) the full contents of the license of JSNetworkX from this page - https://github.com/fkling/JSNetworkX/blob/master/LICENSE.md

Thank you @jrladd, and so sorry for the delay!

**Comment by jrladd** [comment]
No problem, and that all sounds good to me. I'll make these CSS tweaks in the next few days. Thanks!

**Comment by jrladd** [comment]
Cool, this should be ready whenever you are, @esjewett! Let me know if there's any more I can do.

**Comment by simonwiles** [comment]
> I'd like to get @simonwiles input on whether it makes sense to try to convert this to a webpack build and use the jsnetworkx NPM library.

@esjewett -- you mean in preference to just bundling the `jsnetworkx.js` included in this PR using the gulp script?  Given the way that the Palladio core application loads the modules it doesn't make any difference at all, as far as I can see?

`jsnetworkx.js` *does* need to be bundled somehow, though, or this PR ain't gonna work -- @jrladd did you have a plan in this regard?

**Comment by esjewett** [comment]
@simonwiles @jrladd Good point. Currently for testing it looks like it is loaded via a script tag in the index.html, which would have to be done for all Palladio applications. We definitely need a solution to bundle it, as we did for other dependencies for the main Palladio library. That's now been converted to Webpack, so maybe I just need to spend a couple of hours and convert this component to webpack as well. Probably won't be any harder than implementing a new bundling solution in Gulp.

**Comment by simonwiles** [comment]
It just needs `jsnetworkx.js` adding to https://github.com/humanitiesplusdesign/palladio-graph-component/blob/1c1a14879fdca1810122b63217a36e5c2ffb3e11/gulpfile.js#L10 -- and job done, right?

**Comment by esjewett** [comment]
@simonwiles Uh ... yeah 😳 Sorry, I haven't used Gulp in so long now that I didn't even try to think about it :( @jrladd would you be up for making that change as well and testing that it works?

Thank you for the oversight Simon - much needed and appreciated.

**Comment by jrladd** [comment]
@simonwiles @esjewett Yep, I can do that. I'll let you know when it's done and tested. Thanks, both!

**Comment by jrladd** [comment]
Okay @esjewett @simonwiles, I believe this is all set! I tested locally, and it worked fine with the script tag removed from `index.html` and with `jsnetworkx.js` added to the gulpfile instead.

**Comment by simonwiles** [comment]
@jrladd I've merged this PR and cut a new version of this component (`1.2.0`).  I had to make two small changes to make it work properly -- see especially 3a999886c3ad645e462e26be05c391eace8a82cd and b01226333b43fef1cf295c8590ff245fc166196d.  If you could check these out and confirm that they are appropriate, that'd be great.  A staging version of the Palladio site with these changes merged is available for testing at https://palladio-testing.netlify.app/.  If you're happy that this is what you had in mind and that everything is as it should be, I'll deploy these changes to the main site.

Thanks!

**Review by simonwiles — COMMENTED** [review]
Merged.

**Comment by jrladd** [comment]
@simonwiles Thanks; this all looks great!

One thing I noticed: the table headers are supposed to re-sort each column, but instead clicking on them reloads the page. This was working when I tested a while back, but it may be that I missed something in Angular. It could be that the `href="#"` in each table header is overriding `ng-click`, or maybe there needs to be a `preventDefault()` somewhere. Should I try to fix this in my fork?

Also, and maybe this is a question for @esjewett as well, but would it be possible to include an unobtrusive small print message that says the metrics came from [Network Navigator](https://network-navigator.library.cmu.edu/) with a link back to that page? Maybe in the empty space under Global Metrics? We could link to Palladio from our site, too; we've been working on some updates, and it would be cool to have the sites cross-linked. Apologies that I'm just thinking of this now, as I see the final product here.

Thanks for all your help with this!

**Comment by esjewett** [comment]
This would be more of a question for @quinnanya than for me, but I wouldn't expect it would be a problem to tip a hat to the source.

**Comment by simonwiles** [comment]
After a nudge from Quinn today, I've fixed the issue with the column headings, added the JSNetworkX license to the repo, and added an acknowledgement and a link to Network Navigator to the metrics panel.  I've created a `1.2.1` release of this component, and updated and redeployed the main app, so it's now all live.  Just the 22½ months after your initial PR @jrladd  :)

Thanks again!

**Comment by jrladd** [comment]
!!!! So exciting to see this live. Thanks, @simonwiles! And thanks for all the help over the years (!), @esjewett and @quinnanya.

**Pull request merged**